### PR TITLE
🎉 New Source: Zenloop

### PR DESCRIPTION
**Pull request by AlexanderBatoulis**
## What
* Added new source Zenloop
* Participating in hacktoberfest-2021 https://github.com/airbytehq/hacktoberfest-2021/issues/82

## How
Added four streams for this source
1. Surveys
2. Answers
3. Survey Groups
4. Answers Survey Groups

## Recommended reading order
1. bootstrap.md
2. spec.json
3. Schemas under schemas/*.json
4. source.py

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

## Test Runs
unit test
zenloop_unit_test

integration test
zenloop_acceptance_test

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

### CODE REVIEW

**Comment by alafanechere** [comment]
Hey @AlexanderBatoulis , could you please fill the checklist at the bottom of the PR's description so that we can check if it's ready for review? Thanks! 🙏

File: airbyte-integrations/connectors/source-zenloop/acceptance-test-config.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - config_path: "secrets/config.json"
+  basic_read:
+    - config_path: "secrets/config.json"
+      configured_catalog_path: "integration_tests/configured_catalog.json"
+      empty_streams: []
+#      expect_records:
+#        path: "integration_tests/expected_records.txt"
+#        extra_fields: no
+#        exact_order: no
+#        extra_records: yes
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zenloop/acceptance-test-config.yml:22]: can you remove this?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Thanks @AlexanderBatoulis small comment also can you run `./gradlew format`?

**Comment by AlexanderBatoulis** [comment]
> Thanks @AlexanderBatoulis small comment also can you run `./gradlew format`?

done

**Comment by AlexanderBatoulis** [comment]
> Hey @AlexanderBatoulis , could you please fill the checklist at the bottom of the PR's description so that we can check if it's ready for review? Thanks! 🙏

done

**Comment by marcosmarxm** [comment]
Hello @AlexanderBatoulis ! Sorry to not finished your contribution before the date stipulated in the contest. All contributions made before 15-November are eligible to receive the award. We're trying to review your contribution as soon as possible.

**Review by alafanechere — CHANGES_REQUESTED** [review]
Thank you very much @AlexanderBatoulis for this contribution, I have some minor questions and suggestions that you'll find in my review comments.
In addition, we configured a sandbox account for Zenloop and I'm afraid that the integration tests are not passing when we fill the `public_hash_id` field. The `https://app.zenloop.com/settings/api` page provides list of hash ids for surveys, but this value is reused to build requests to `survey_groups` endpoint in the `AnswersSurveyGroup` stream (`survey_groups/{self.public_hash_id}/answers`). It looks like the API expects a survey group id that is different from the public hash id provided in the config (which is expected to be a *survey id* and not a *survey group id*) and it leads to 404 error.
**TLDR: integration tests did not pass when setting `public_hash_id` to a valid value in our config.**

Could you also fix conflicts with master please? 🙏

File: airbyte-integrations/connectors/source-zenloop/source_zenloop/source.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        if next_page_token:
+            params.update(**next_page_token)
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        yield response_json
+
+
+class ChildStreamMixin:
```
[the remark this answers is not in the document]
Comment (AlexanderBatoulis) [reply airbyte-integrations/connectors/source-zenloop/source_zenloop/source.py:63]: Set up of class to be consistent with [trello](https://github.com/airbytehq/airbyte/blob/f6d81ff8813b8bd1a5cd9923093d5472a8407d76/airbyte-integrations/connectors/source-trello/source_trello/source.py) and [onesignal](https://github.com/airbytehq/airbyte/blob/760f429959075605dbe25adeb8d29d9efff0cc7a/airbyte-integrations/connectors/source-onesignal/source_onesignal/streams.py) connectors.

File: airbyte-integrations/connectors/source-zenloop/source_zenloop/source.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            # if looped through all slices take its date_from parameter
+            # else no public_hash_id provided -> take cursor_field
+            if stream_slice:
+                params["date_from"] = stream_slice["date_from"]
+            else:
+                params["date_from"] = stream_state[self.cursor_field]
+        return params
+
+
+class Surveys(ZenloopStream):
```
[the remark this answers is not in the document]
Comment (AlexanderBatoulis) [reply airbyte-integrations/connectors/source-zenloop/source_zenloop/source.py:120]: Thanks for the suggestion, I added it to it

**Comment by AlexanderBatoulis** [comment]
> Thank you very much @AlexanderBatoulis for this contribution, I have some minor questions and suggestions that you'll find in my review comments. In addition, we configured a sandbox account for Zenloop and I'm afraid that the integration tests are not passing when we fill the `public_hash_id` field. The `https://app.zenloop.com/settings/api` page provides list of hash ids for surveys, but this value is reused to build requests to `survey_groups` endpoint in the `AnswersSurveyGroup` stream (`survey_groups/{self.public_hash_id}/answers`). It looks like the API expects a survey group id that is different from the public hash id provided in the config (which is expected to be a _survey id_ and not a _survey group id_) and it leads to 404 error. **TLDR: integration tests did not pass when setting `public_hash_id` to a valid value in our config.**
> 
> Could you also fix conflicts with master please? 🙏

Hi @alafanechere,
Thank you very much for your thorough review! I resolved the not passing integration tests by adding another request parameter for survey_group_id. 
Would be great if you could take another look and let me know if there is anything outstanding.

Kind regards,
Alex

**Comment by alafanechere** [comment]
HI @AlexanderBatoulis , thanks for your modifications, I'll go for another review.

**Review by alafanechere — CHANGES_REQUESTED** [review]
Hi @AlexanderBatoulis , I tried to run the acceptance tests but they are failing. Could you please make sure  the following command runs flawlessly?
`./gradlew :airbyte-integrations:connectors:source-zenloop:integrationTest`

**Comment by AlexanderBatoulis** [comment]
Hi @alafanechere, this should run fine now - missed to update setup.py

**Review by alafanechere — APPROVED** [review]
LGTM, I'm happy with the change you've made and successfully ran the acceptance tests!
Ready to merge 👏
